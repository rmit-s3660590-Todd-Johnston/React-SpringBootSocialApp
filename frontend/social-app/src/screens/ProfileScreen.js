import React, {Component} from 'react';
import {
    Avatar,
    Descriptions,
    Divider,
    Button,
    Card,
    Badge,
    Icon,
    Typography,
    Layout,
    Modal,
    Input,
} from "antd";
import AuthenticationService from "../AuthenticationService";
import UserBeanService from "../api/UserBeanService";

const gridStyle = {
    width: '25%',
    textAlign: 'center',
    backgroundColor: 'lightgray',
};

export default class ProfileScreen extends Component {
    constructor(props){
        super(props);
        this.state = {
            userID: undefined,
            userName: 'Temp Name',
            firstName: 'First',
            lastName: 'Last',
            profilePic: undefined,
            isMentor: false,
            subjects: [],
            modalVisible: false,
            editProfileText: ''
        };
    };


//     id: 666
// ​
//     last_name: "error"
// ​
//     mentor: false
// ​
//     name: "error"
// ​ ​
//     profilePic: "https://short-biography.com/wp-content/uploads/mark-zuckerberg/Mark-Zuckerberg-300x300.jpg"
// ​
//     subjects: Array(4) [ null, null, null, … ]


    getLoggedInUserData = () => {
        console.log("Logged in user: " + AuthenticationService.getLoggedInUserName());
        UserBeanService.retrieveUserBeanByUserName(AuthenticationService.getLoggedInUserName())
            .then((res) => {
                console.log(res.data);
                console.log(res.data.subjects);
                this.setState({
                    userID: res.data.id,
                    userName: res.data.user_name,
                    firstName: res.data.name,
                    lastName: res.data.last_name,
                    profilePic: res.data.profilePic,
                    isMentor: res.data.mentor,
                    subjects: res.data.subjects
                });
            })
            .catch(err => console.error(err));
    };

    nullError(){
        //subject array
        console.log("subject state: "+ this.state.subjects[0]);
        let i;
        for (i = 0; i < this.state.subjects.length-1; i++) {
            if (this.state.subjects[i] === undefined){
                this.state.subjects[i] = "error";
            }
        }
        console.log("subject state: "+ this.state.subjects[0]);

        //mentor
        if (this.state.isMentor === undefined){
            this.setState({isMentor: true});
        }
    };

    showEditProfile = () =>{
        this.setState({
            modalVisible: true,
        });
    };

    handleOk = e => {
        let user = {
            id: this.state.userID,
            profiePic: this.state.editProfileText
        };
        console.log(e);
        UserBeanService.updateUserBean(this.state.userID , user)
            .then((res) => {
                console.log("User: " + res.data.id + " updated!");
            })
            .catch(res => console.log(res));

        this.setState({
            modalVisible: false,
        });
        this.forceUpdate();
    };

    handleCancel = e => {
        console.log(e);
        this.setState({
            modalVisible: false,
        });
    };

    handleValueChange = e => {
        console.log(e.target.value);
        this.setState({editProfileText: e.target.value})
    };

    componentDidMount() {
        this.getLoggedInUserData();
        this.nullError();
    }

    render() {
        return <Layout.Content style={{backgroundColor: "#FFFFFF", margin: 13, padding: 13}}>
            <Typography.Title>Profile</Typography.Title>
            <div style={{display: 'flex', justifyContent: 'center', alignItems: 'center', height: '10vh'}}>
                <Badge id={'bigBadge'} count={<Icon style={{marginBottom: '10px', color: 'gray'}} type={"edit"} onClick={this.showEditProfile}/>} >
                    <Avatar src={this.state.profilePic}
                            size={128} icon="user"/>
                </Badge>
            </div>

            <Modal
                visible ={this.state.modalVisible}
                title="Edit Profile Picture"
                okText="Change"
                onCancel={this.handleCancel}
                onOk={this.handleOk}
            >
                <Input placeholder={"Input link here: "} onChange={value => this.handleValueChange(value)}/>

            </Modal>

            <div style={{display: 'flex', justifyContent: 'center', alignItems: 'center', height: '10vh'}}>
                <h1>{this.state.firstName + " " + this.state.lastName}</h1>
            </div>

            <div style={{display: 'flex', justifyContent: 'center', alignItems: 'center'}}>
                <Button style={{marginRight: 60}} size={'large'}
                        onClick={() => {this.props.history.push('/wall')}}
                >
                    Go To Wall
                </Button>
                <Button style={{marginRight: 60}} type='primary' size={'large'}
                        onClick={() => this.props.history.push('/chat')}
                >
                    Message Me
                </Button>
                <Button style={{marginRight: 60}} type='primary' size={'large'}
                        onClick={() => this.props.history.push('/study-group')}
                >
                    Add to study group
                </Button>
                <Button size={'large'}>Add Friend?</Button>
            </div>

            <Divider style={{color: 'black', font: 'bold', marginTop: '30px'}}>About Me</Divider>
            <Descriptions>
                <Descriptions.Item label="First Name">{this.state.firstName}</Descriptions.Item>
                <Descriptions.Item label="Last Name">{this.state.lastName}</Descriptions.Item>
                <Descriptions.Item label="Mentor status">{this.state.isMentor.toString()}</Descriptions.Item>
                <Descriptions.Item label="User ID: "> {this.state.userID}</Descriptions.Item>
            </Descriptions>

            <Divider style={{color: 'black', font: 'bold', marginTop: '30px'}}>My Skills</Divider>
            <Card.Grid style={gridStyle}>{this.state.subjects[0]}</Card.Grid>
            <Card.Grid style={gridStyle}>{this.state.subjects[1]}</Card.Grid>
            <Card.Grid style={gridStyle}>{this.state.subjects[2]}</Card.Grid>
            <Card.Grid style={gridStyle}>{this.state.subjects[3]}</Card.Grid>
        </Layout.Content>
    }
}