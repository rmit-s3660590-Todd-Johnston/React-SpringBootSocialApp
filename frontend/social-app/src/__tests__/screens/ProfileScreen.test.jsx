import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM, {render, unmountComponentAtNode} from 'react-dom';
import {act, mockComponent} from "react-dom/test-utils";
import ProfileScreen from "../../screens/ProfileScreen";
import {shallow, mount} from 'enzyme';

let container = null;
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
});


it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<ProfileScreen/>,div) ;
    ReactDOM.unmountComponentAtNode(div);
});

describe("<ProfileScreen/>", () => {
    it('edit profile not shown at start', () => {
        act(() => {
            render(<ProfileScreen/>, container)
        });

        expect(container).toMatchSnapshot();
    });

    let wrapper = shallow(<ProfileScreen/>);
    console.log("modal visible " + wrapper.state().modalVisible);

    expect(wrapper.state().modalVisible).toEqual(false);

    it("modal isn't shown", () => {
        act(() => {
            render(<ProfileScreen/>, container)
        });

        expect(container).toMatchSnapshot();
        expect(wrapper.state().modalVisible).toBe(false);

    });

    it("modal is shown after click", () => {
        act(() => {
            render(<ProfileScreen/>, container)
        });
        const profileScreen = mount(<ProfileScreen/>)

        expect(container).toMatchSnapshot();
        const button = profileScreen.find('[data-testid="edit-button"]');
        button.at(0).simulate('click', profileScreen.setState({modalVisible: true}));
        expect(profileScreen.state().modalVisible).toBe(true);
        //this should be true

    });

    it('should return changed name', () => {
        render(<ProfileScreen/>, container)
    });
    const profileScreen = mount(<ProfileScreen/>)
    expect(container).toMatchSnapshot();
    const firstName = profileScreen.find('[data-testid="test-name"]');
    profileScreen.setState({firstName: 'Test', lastName: 'Last'})
    expect(firstName.text()).toBe('Test Last');


    it('edit profile text is empty', () => {
        let wrapper = shallow(<ProfileScreen/>);
        console.log("modal visible " + wrapper.state().editProfileText);

        expect(wrapper.state().editProfileText).toEqual('');
    });
});

//
//     it('edit profile button is clickable', () => {
//         let wrapper = shallow(<ProfileScreen/>);
//         const div = document.createElement('div');
//         document.body.appendChild(div);
//
//         expect(wrapper.state().modalVisible).toEqual(false);
//
//         act(() => {
//             let wrapper = shallow(<ProfileScreen/>, div);
//         });
//
//         const badge = document.querySelector("[data-testid=edit-button]");
//
//         act(() => {
//             badge.dispatchEvent(new MouseEvent("click"))
//         });
//         //if no error button is clickable
//     });
//
//
//
//     it('wall button click', () => {
//         const div = document.createElement('div');
//         document.body.appendChild(div);
//
//         act(() => {
//             ReactDOM.render(<ProfileScreen/>, div);
//         });
//
//         const button = document.querySelector("[data-testid=test-wall]");
//
//         act(() => {
//             button.dispatchEvent(new MouseEvent("click"))
//         });
//         //if no error button is clickable
//     });
//
//     it('message button click', () => {
//         const div = document.createElement('div');
//         document.body.appendChild(div);
//
//         act(() => {
//             ReactDOM.render(<ProfileScreen/>, div);
//         });
//
//         const button = document.querySelector("[data-testid=test-message]");
//
//         act(() => {
//             button.dispatchEvent(new MouseEvent("click"))
//         });
//         //if no error button is clickable
//     });
//
//     it('study group button click', () => {
//         const div = document.createElement('div');
//         document.body.appendChild(div);
//
//         act(() => {
//             ReactDOM.render(<ProfileScreen/>, div);
//         });
//
//         const button = document.querySelector("[data-testid=test-study]");
//
//         act(() => {
//             button.dispatchEvent(new MouseEvent("click"))
//         });
//         //if no error button is clickable
//     });
//
// });
// ;