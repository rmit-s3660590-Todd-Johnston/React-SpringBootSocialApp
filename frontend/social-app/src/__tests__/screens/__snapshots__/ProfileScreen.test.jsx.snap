// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileScreen/> modal is shown after click 1`] = `
<div>
  <main
    class="ant-layout-content"
    style="background-color: rgb(255, 255, 255); margin: 13px; padding: 13px;"
  >
    <h1
      class="ant-typography"
    >
      Profile
    </h1>
    <div
      style="display: flex; justify-content: center; align-items: center; height: 10vh;"
    >
      <span
        class="ant-badge"
        data-testid="edit-button"
      >
        <span
          class="ant-avatar ant-avatar-circle ant-avatar-icon"
          style="width: 128px; height: 128px; line-height: 128px; font-size: 64px;"
        >
          <i
            aria-label="icon: user"
            class="anticon anticon-user"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="user"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
              />
            </svg>
          </i>
        </span>
        <i
          aria-label="icon: edit"
          class="anticon anticon-edit ant-scroll-number-custom-component ant-badge-zoom-appear"
          style="margin-bottom: 10px; color: gray;"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="edit"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
            />
          </svg>
        </i>
      </span>
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; height: 10vh;"
    >
      <h1>
        First Last
      </h1>
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center;"
    >
      <button
        class="ant-btn ant-btn-lg"
        data-testid="test-wall"
        style="margin-right: 60px;"
        type="button"
      >
        <span>
          Go To Wall
        </span>
      </button>
      <button
        class="ant-btn ant-btn-primary ant-btn-lg"
        data-testid="test-message"
        style="margin-right: 60px;"
        type="button"
      >
        <span>
          Message Me
        </span>
      </button>
      <button
        class="ant-btn ant-btn-primary ant-btn-lg"
        data-testid="test-study"
        style="margin-right: 60px;"
        type="button"
      >
        <span>
          Add to study group
        </span>
      </button>
      <button
        class="ant-btn ant-btn-lg"
        type="button"
      >
        <span>
          Add Friend?
        </span>
      </button>
    </div>
    <div
      class="ant-divider ant-divider-horizontal ant-divider-with-text-center"
      role="separator"
      style="color: black; font: bold bold; margin-top: 30px;"
    >
      <span
        class="ant-divider-inner-text"
      >
        About Me
      </span>
    </div>
    <div
      class="ant-descriptions"
    >
      <div
        class="ant-descriptions-view"
      >
        <table>
          <tbody>
            <tr
              class="ant-descriptions-row"
            >
              <td
                class="ant-descriptions-item"
                colspan="1"
              >
                <span
                  class="ant-descriptions-item-label ant-descriptions-item-colon"
                >
                  First Name
                </span>
                <span
                  class="ant-descriptions-item-content"
                >
                  First
                </span>
              </td>
              <td
                class="ant-descriptions-item"
                colspan="1"
              >
                <span
                  class="ant-descriptions-item-label ant-descriptions-item-colon"
                >
                  Last Name
                </span>
                <span
                  class="ant-descriptions-item-content"
                >
                  Last
                </span>
              </td>
              <td
                class="ant-descriptions-item"
                colspan="1"
              >
                <span
                  class="ant-descriptions-item-label ant-descriptions-item-colon"
                >
                  Mentor status
                </span>
                <span
                  class="ant-descriptions-item-content"
                >
                  false
                </span>
              </td>
            </tr>
            <tr
              class="ant-descriptions-row"
            >
              <td
                class="ant-descriptions-item"
                colspan="3"
              >
                <span
                  class="ant-descriptions-item-label ant-descriptions-item-colon"
                >
                  User ID: 
                </span>
                <span
                  class="ant-descriptions-item-content"
                >
                   
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="ant-divider ant-divider-horizontal ant-divider-with-text-center"
      role="separator"
      style="color: black; font: bold bold; margin-top: 30px;"
    >
      <span
        class="ant-divider-inner-text"
      >
        My Skills
      </span>
    </div>
    <div
      class="ant-card-grid"
      style="width: 25%; text-align: center; background-color: lightgray;"
    />
    <div
      class="ant-card-grid"
      style="width: 25%; text-align: center; background-color: lightgray;"
    />
    <div
      class="ant-card-grid"
      style="width: 25%; text-align: center; background-color: lightgray;"
    />
    <div
      class="ant-card-grid"
      style="width: 25%; text-align: center; background-color: lightgray;"
    />
  </main>
</div>
`;

exports[`<ProfileScreen/> modal isn't shown 1`] = `
<div>
  <main
    class="ant-layout-content"
    style="background-color: rgb(255, 255, 255); margin: 13px; padding: 13px;"
  >
    <h1
      class="ant-typography"
    >
      Profile
    </h1>
    <div
      style="display: flex; justify-content: center; align-items: center; height: 10vh;"
    >
      <span
        class="ant-badge"
        data-testid="edit-button"
      >
        <span
          class="ant-avatar ant-avatar-circle ant-avatar-icon"
          style="width: 128px; height: 128px; line-height: 128px; font-size: 64px;"
        >
          <i
            aria-label="icon: user"
            class="anticon anticon-user"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="user"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
              />
            </svg>
          </i>
        </span>
        <i
          aria-label="icon: edit"
          class="anticon anticon-edit ant-scroll-number-custom-component ant-badge-zoom-appear"
          style="margin-bottom: 10px; color: gray;"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="edit"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
            />
          </svg>
        </i>
      </span>
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; height: 10vh;"
    >
      <h1>
        First Last
      </h1>
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center;"
    >
      <button
        class="ant-btn ant-btn-lg"
        data-testid="test-wall"
        style="margin-right: 60px;"
        type="button"
      >
        <span>
          Go To Wall
        </span>
      </button>
      <button
        class="ant-btn ant-btn-primary ant-btn-lg"
        data-testid="test-message"
        style="margin-right: 60px;"
        type="button"
      >
        <span>
          Message Me
        </span>
      </button>
      <button
        class="ant-btn ant-btn-primary ant-btn-lg"
        data-testid="test-study"
        style="margin-right: 60px;"
        type="button"
      >
        <span>
          Add to study group
        </span>
      </button>
      <button
        class="ant-btn ant-btn-lg"
        type="button"
      >
        <span>
          Add Friend?
        </span>
      </button>
    </div>
    <div
      class="ant-divider ant-divider-horizontal ant-divider-with-text-center"
      role="separator"
      style="color: black; font: bold bold; margin-top: 30px;"
    >
      <span
        class="ant-divider-inner-text"
      >
        About Me
      </span>
    </div>
    <div
      class="ant-descriptions"
    >
      <div
        class="ant-descriptions-view"
      >
        <table>
          <tbody>
            <tr
              class="ant-descriptions-row"
            >
              <td
                class="ant-descriptions-item"
                colspan="1"
              >
                <span
                  class="ant-descriptions-item-label ant-descriptions-item-colon"
                >
                  First Name
                </span>
                <span
                  class="ant-descriptions-item-content"
                >
                  First
                </span>
              </td>
              <td
                class="ant-descriptions-item"
                colspan="1"
              >
                <span
                  class="ant-descriptions-item-label ant-descriptions-item-colon"
                >
                  Last Name
                </span>
                <span
                  class="ant-descriptions-item-content"
                >
                  Last
                </span>
              </td>
              <td
                class="ant-descriptions-item"
                colspan="1"
              >
                <span
                  class="ant-descriptions-item-label ant-descriptions-item-colon"
                >
                  Mentor status
                </span>
                <span
                  class="ant-descriptions-item-content"
                >
                  false
                </span>
              </td>
            </tr>
            <tr
              class="ant-descriptions-row"
            >
              <td
                class="ant-descriptions-item"
                colspan="3"
              >
                <span
                  class="ant-descriptions-item-label ant-descriptions-item-colon"
                >
                  User ID: 
                </span>
                <span
                  class="ant-descriptions-item-content"
                >
                   
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="ant-divider ant-divider-horizontal ant-divider-with-text-center"
      role="separator"
      style="color: black; font: bold bold; margin-top: 30px;"
    >
      <span
        class="ant-divider-inner-text"
      >
        My Skills
      </span>
    </div>
    <div
      class="ant-card-grid"
      style="width: 25%; text-align: center; background-color: lightgray;"
    />
    <div
      class="ant-card-grid"
      style="width: 25%; text-align: center; background-color: lightgray;"
    />
    <div
      class="ant-card-grid"
      style="width: 25%; text-align: center; background-color: lightgray;"
    />
    <div
      class="ant-card-grid"
      style="width: 25%; text-align: center; background-color: lightgray;"
    />
  </main>
</div>
`;
